### PR TITLE
feature/magnet-finder

### PR DESCRIPTION
**Pull request by RandomTanne**

### CODE REVIEW

File: fluminum/app/(tabs)/magnet-finder.tsx [hunk 1]
```diff
@@ -1,24 +1,74 @@
...
+
+  function calcSoundRate(absoluteMagnetometerValue:number) {
+    return 1 + ((4*absoluteMagnetometerValue)/1000)
+  }
 
   React.useEffect(() => {
-    const l = Magnetometer.addListener((e) =>
-      setDatas((d) => [...d, e].slice(-5)),
-    );
+    if (sound == undefined) return;
```
Comment (c0rydoras) [inline fluminum/app/(tabs)/magnet-finder.tsx]: use `===` :)

**Review by c0rydoras — APPROVED** [review]

File: LICENSE [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
```
Comment (c0rydoras) [inline LICENSE]: this looks wrong

only change line 3 (copyright) plesae

**Review by c0rydoras — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -3,3 +3,7 @@
 ## License
 
 **Code** released under the [MIT License](LICENSE).
+
+### Attributions
```
Comment (c0rydoras) [inline README.md]:
```suggestion
## Attribution
```

File: fluminum/app/(tabs)/magnet-finder.tsx [hunk 4]
```diff
@@ -1,24 +1,81 @@
...
+        <View className="flex-1 justify-center items-center gap-5 p-6 bg-secondary/30">
+          <AnimatedCircularProgress
+              size={300}
+              rotation={270}
+              width={15}
+              arcSweepAngle={180}
+              fill={magnetometerValue/10}
+              tintColor="#FFE500"
+              backgroundColor="#3d5875" />
+          <Text className="text-muted-foreground text-6xl">{`${magnetometerValue.toString()}µT`}</Text>
```
Comment (c0rydoras) [inline fluminum/app/(tabs)/magnet-finder.tsx]:
```suggestion
          <Text className="text-muted-foreground text-6xl">{`${magnetometerValue.toFixed(2)}µT`}</Text>
```

File: fluminum/app/(tabs)/magnet-finder.tsx [hunk 5]
```diff
@@ -1,24 +1,81 @@
...
   React.useEffect(() => {
-    const l = Magnetometer.addListener((e) =>
-      setDatas((d) => [...d, e].slice(-5)),
-    );
+    if (sound === undefined) return;
+
+    let time = 0;
+    let oldSoundRate = 1;
+    const listener = Magnetometer.addListener(async (e) => {
+        const absoluteMagnetometerValue = Number((Math.sqrt((e.x**2) + (e.y**2) + (e.z**2)).toFixed(2)));
```
Comment (c0rydoras) [inline fluminum/app/(tabs)/magnet-finder.tsx]:
```suggestion
        const absoluteMagnetometerValue = Math.sqrt((e.x**2) + (e.y**2) + (e.z**2));
```

File: fluminum/app/(tabs)/magnet-finder.tsx [hunk 6]
```diff
@@ -1,24 +1,81 @@
...
 
   React.useEffect(() => {
-    const l = Magnetometer.addListener((e) =>
-      setDatas((d) => [...d, e].slice(-5)),
-    );
+    if (sound === undefined) return;
+
+    let time = 0;
+    let oldSoundRate = 1;
+    const listener = Magnetometer.addListener(async (e) => {
```
Comment (c0rydoras) [inline fluminum/app/(tabs)/magnet-finder.tsx]: consider naming this something like `data` or `measurement` instead of `e`

**Review by c0rydoras — CHANGES_REQUESTED** [review]

**Review by c0rydoras — APPROVED** [review]

**Review by MarklinH0 — APPROVED** [review]

**Pull request merged**